### PR TITLE
[nitro-protocol] Upgrade solidity to 0.7.4

### LINKED ISSUE #2870: Bump solidity version to 0.7

### PR DESCRIPTION
**Pull request by geoknee**
~⚠️ base is not master, see #2893~

Closes #2870

### CODE REVIEW

File: packages/devtools/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "lockfile": "1.0.4",
     "pino": "6.2.0",
     "pino-pretty": "4.0.0",
-    "solc": "0.6.12",
+    "solc": "0.7.4",
```
Comment (geoknee) [inline packages/devtools/package.json:23]: I suspect we no longer need to install `solc` (anywhere?)

File: .circleci/config.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ defaults: &defaults
   working_directory: /home/circleci/project
   resource_class: medium
   docker:
-    - image: geoknee/statechannels:12.18-solc0.6.12 # solc installed (unecessary since hardhat will re-download solc)
+    - image: geoknee/statechannels:12.18-solc0.7.4 # solc installed (unecessary since hardhat will re-download solc)
```
Comment (geoknee) [inline .circleci/config.yml:14]: This is harmless but probably unnecessary given our use of hardhat for compilation.

File: packages/nitro-protocol/contracts/ForceMove.sol [hunk 3]
```diff
@@ -84,7 +84,7 @@ contract ForceMove is IForceMove {
         emit ChallengeRegistered(
             channelId,
             largestTurnNum,
-            uint48(now) + fixedPart.challengeDuration, // This could overflow, so don't join a channel with a huge challengeDuration
+            uint48(block.timestamp) + fixedPart.challengeDuration, // This could overflow, so don't join a channel with a huge challengeDuration
```
Comment (snario) [inline packages/nitro-protocol/contracts/ForceMove.sol:87]: What's the max `challengeDuration`? We could add a check to the wallet

**Review by snario — APPROVED** [review]

File: packages/nitro-protocol/contracts/ForceMove.sol (hunk 3 above)
Comment (geoknee) [reply packages/nitro-protocol/contracts/ForceMove.sol:87]: I think (2**48 − 1 - now) but I would need to think a bit more to be sure. But I think this is probably the safe lower bound on the maximum.

File: packages/devtools/package.json (hunk 1 above)
Comment (geoknee) [reply packages/devtools/package.json:23]: #2902

**Pull request merged**